include/sound/sof/virtio.h

```diff
@@ -0,0 +1,107 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+ *
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2018-2019 Intel Corporation. All rights reserved.
+ *
+ *  Contact Information:
+ *  Author:	Luo Xionghu <xionghu.luo@intel.com>
+ *		Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ *		Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>
+ */
+
+#ifndef _SOF_VIRTIO_H
+#define _SOF_VIRTIO_H
+
+#include <sound/sof/header.h>
+
+/*
+ * Currently we define 3 vqs: one for handling of IPC commands, one for
+ * handling of stream position updates, and one for audio data.
+ */
+enum {
+	SOF_VIRTIO_IPC_CMD_VQ,	/* IPC commands and replies */
+	SOF_VIRTIO_IPC_PSN_VQ,	/* Stream position updates */
+	SOF_VIRTIO_DATA_VQ,	/* Audio data */
+	/* Keep last */
+	SOF_VIRTIO_NUM_OF_VQS,
```

Comment:
@lyakh this is weird. "NUM" as part of the enum. Instead I'd say just call is SOF_VIRTIO_LAST and point it to SOF_VIRTIO_DATA_VQ and it should serve the same purpose no?

Reply:
@ranj063 It does look weird to me too but I think there are other instances of this particular pattern that I have tripped upon (for example, `enum snd_sof_fw_blk_type` in `include/uapi/sound/sof/fw.h`).

Reply:
@ranj063 it isn't that weird IMHO, you can find this used in multiple locations. Sure, you could do LAST, but then if the last element changes, you'd also need to change this line. And since I don't really need the last element, but their number, I'd need another macro to do `#define NUM (LAST - FIRST + 1)` and then if FIRST changes, I'd have to change that too...

Reply:
...even assuming `FIRST == 0` still I don't see an advantage in this your proposed change